### PR TITLE
feat(cdk): switch injectables to new scope API

### PR DESCRIPTION
**Pull request by jelbourn**

### CODE REVIEW

File: src/cdk/overlay/overlay-directives.ts [hunk 1]
```diff
@@ -55,21 +63,18 @@ const defaultPositionList = [
...
-
-/** @docs-private */
-export const CDK_CONNECTED_OVERLAY_SCROLL_STRATEGY_PROVIDER = {
-  provide: CDK_CONNECTED_OVERLAY_SCROLL_STRATEGY,
-  deps: [Overlay],
-  useFactory: CDK_CONNECTED_OVERLAY_SCROLL_STRATEGY_PROVIDER_FACTORY,
-};
-
+    new InjectionToken<() => ScrollStrategy>('cdk-connected-overlay-scroll-strategy', {
+  scope: APP_ROOT_SCOPE,
```
Comment (mmalerba) [inline src/cdk/overlay/overlay-directives.ts]: nit: indentation

File: src/cdk/platform/platform-module.ts [hunk 2]
```diff
@@ -7,10 +7,7 @@
  */
 
 import {NgModule} from '@angular/core';
-import {Platform} from './platform';
 
 
-@NgModule({
-  providers: [Platform]
-})
+@NgModule()
```
Comment (mmalerba) [inline src/cdk/platform/platform-module.ts]: Out of curiosity, will we be able to kill this module off in the future? It appears to do nothing now

File: src/cdk/a11y/a11y-module.ts [hunk 3]
```diff
@@ -9,27 +9,12 @@
...
 @NgModule({
   imports: [CommonModule, PlatformModule],
   declarations: [CdkTrapFocus, FocusTrapDeprecatedDirective, CdkMonitorFocus],
   exports: [CdkTrapFocus, FocusTrapDeprecatedDirective, CdkMonitorFocus],
-  providers: [
-    InteractivityChecker,
-    FocusTrapFactory,
-    AriaDescriber,
-    LIVE_ANNOUNCER_PROVIDER,
-    ARIA_DESCRIBER_PROVIDER,
```
Comment (crisbeto) [inline src/cdk/a11y/a11y-module.ts]: I'm not super familiar with how the app-wide providers work yet, but should the `AriaDescriber` still be set in the `providers` in this case? I believe that it's the only reason why, for example, the tooltip imports the `A11yModule`.

File: src/cdk/bidi/dir-document-token.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ * We also can't re-provide the DOCUMENT token from platform-brower because the unit tests
+ * themselves use things like `querySelector` in test code.
+ *
+ * This token is defined in a separate file from Directionality as a workaround for
+ * https://github.com/angular/angular/issues/22559
+ *
+ * @docs-private
+ */
+export const DIR_DOCUMENT = new InjectionToken<Document>('cdk-dir-doc', {
```
Comment (crisbeto) [inline src/cdk/bidi/dir-document-token.ts]: As an aside, do we really need a separate token for the document here? We can easily stub out the `DOCUMENT` itself in the tests.

File: src/cdk/overlay/overlay-directives.ts (hunk 1 above)
Comment (jelbourn) [reply src/cdk/overlay/overlay-directives.ts]: Done

File: src/cdk/platform/platform-module.ts (hunk 2 above)
Comment (jelbourn) [reply src/cdk/platform/platform-module.ts]: I suppose so for now, though we could add other stuff to it. Maybe something like
`*cdkIfPlatform="IE11, Safari"`

File: src/cdk/a11y/a11y-module.ts (hunk 3 above)
Comment (jelbourn) [reply src/cdk/a11y/a11y-module.ts]: As I understand it, module imports just for this kind of injectable shouldn't be necessary any more. I'll address those in my next PR that applies this to material.

File: src/cdk/bidi/dir-document-token.ts (hunk 4 above)
Comment (jelbourn) [reply src/cdk/bidi/dir-document-token.ts]: When I was originally doing this I found that re-providing common's `DOCUMENT` with something else made Angular not work.

**Comment by jelbourn** [comment]
I've re-added the old providers because some people were depending on them

**Pull request merged**